app/proc/telegram.go

```diff
@@ -48,18 +51,97 @@ func (client TelegramClient) Send(channelID string, item feed.Item) error {
 		return nil
 	}
 
+	contentLength, err := getContentLength(item.Enclosure.URL)
+	if err != nil {
+		return err
+	}
+
+	var message *tb.Message
+
+	if contentLength < 50_000_000 {
+		message, err = client.sendAudio(channelID, item)
+	} else {
+		message, err = client.sendText(channelID, item)
+	}
+
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] send telegram message: \n%s", message.Text)
+	return err
+}
+
+func getContentLength(url string) (int64, error) {
```

Comment:
@umputun how mocking http response?

Reply:
see httptest. You can run test server and provide whatever response you need

Reply:
this is an example of httptest.Server use in real life, from one of my projects https://github.com/go-pkgz/rest/blob/master/httperrors_test.go#L16